controllers/common.go

```diff
@@ -57,6 +58,74 @@ func BuildCollector(params manifests.Params) ([]client.Object, error) {
 	return resources, nil
 }
 
+// @TODO: Later merge this in to reconcileDesiredObjects after implementing search functions for DCGM
+func reconcileDesiredObjectsWPrune(ctx context.Context, kubeClient client.Client, logger logr.Logger, owner v1alpha1.AmazonCloudWatchAgent, scheme *runtime.Scheme,
+	desiredObjects []client.Object,
+	searchOwnedObjectsFunc func(ctx context.Context, owner v1alpha1.AmazonCloudWatchAgent) (map[types.UID]client.Object, error),
+) error {
+	var errs []error
+	existingObjectMap := make(map[types.UID]client.Object)
+	existingObjectList := []client.Object{}
+	previouslyOwnedObjects, err := searchOwnedObjectsFunc(ctx, owner)
+	if err != nil {
+		return err
```

Comment:
Can we improve on the error message here? I'm not sure what it would look like from a customer's perspective.